internal/prometheusbp/spectest/spec_test.go

```diff
@@ -0,0 +1,326 @@
+			prefix:         "test",
+			clientOrServer: server,
+			want:           map[string]struct{}{"test_method": {}, "test_service": {}, "test_success": {}},
+		},
+		{
+			name:           "requests_total labels",
+			metricName:     "test_requests_total",
+			prefix:         "test",
+			clientOrServer: client,
+			want: map[string]struct{}{
+				"test_method":                {},
+				"test_service":               {},
+				"test_success":               {},
+				"test_baseplate_status":      {},
+				"test_baseplate_status_code": {},
+				"test_exception_type":        {},
+				"test_slug":                  {},
+			},
+		},
+		{
+			name:           "active_requests labels",
+			metricName:     "test_active_requests",
+			prefix:         "test",
+			clientOrServer: server,
+			want:           map[string]struct{}{"test_method": {}, "test_service": {}},
+		},
+		{
+			name:           "none",
+			prefix:         "test",
+			clientOrServer: server,
+			want:           map[string]struct{}{},
+		},
+	}
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			if got, want := buildLables(tt.metricName, tt.prefix, tt.clientOrServer), tt.want; !reflect.DeepEqual(got, want) {
+				t.Fatalf("got %v, want %v", got, want)
+			}
+		})
+	}
+}
+
+func TestValidateSpec(t *testing.T) {
+	var (
+		testLabels = []string{
+			"thrift_method",
+			"thrift_service",
+			"thrift_success",
+			"thrift_baseplate_status",
+		}
+
+		testMetric = promauto.NewCounterVec(prometheus.CounterOpts{
+			Name: "thrift_client_latency_seconds",
+			Help: "Test help message",
+		}, testLabels)
+	)
+
+	labels := prometheus.Labels{
+		"thrift_method":           "foo",
+		"thrift_service":          "foo",
+		"thrift_success":          "foo",
+		"thrift_baseplate_status": "foo",
+	}
+	testMetric.With(labels).Inc()
+
+	testCases := []struct {
+		name           string
+		metric         prometheus.Collector
+		prefix         string
+		clientOrServer string
+		wantCount      int
+		wantErrs       []error
+	}{
+		{
+			name:           "not found server",
+			metric:         testMetric,
+			prefix:         "thrift",
+			clientOrServer: "server",
+			wantErrs:       []error{errNotFound},
+		},
+		{
+			name:           "multi errs client",
+			metric:         testMetric,
+			prefix:         "thrift",
+			clientOrServer: "client",
+			wantErrs:       []error{errPrometheusLint, errDiffLabels, errNotFound},
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			gotErr := validateSpec(tt.prefix, tt.clientOrServer)
+			checkErrors(t, gotErr, tt.wantErrs)
+		})
+	}
+}
+
+func checkErrors(tb testing.TB, gotErr error, wantErrs []error) {
+	tb.Helper()
+
+	if got, want := errorsbp.BatchSize(gotErr), len(wantErrs); got != want {
+		tb.Errorf("not same number of errors got %d, want %d", got, want)
+	}
+
+	for _, wantErr := range wantErrs {
+		if !errors.Is(gotErr, wantErr) {
+			tb.Fatalf("error mismatch: got %v, want %v", gotErr, wantErr)
```

Comment:
1. this should be `Errorf` instead of `Fatalf`
2. this is not really a got/want situation. the error message should be something like this: `tb.Errorf("Want error %v not in returned errors", wantErr)` (I did not add want into the error message because the `tb.Logf` part above would print that info).